tensorflow/python/ops/image_grad_deterministic_test.py

```diff
@@ -127,6 +135,164 @@ def resize_bilinear_gradients(local_seed):
           self.assertAllEqual(result_a, result_b)
 
 
+class CropAndResizeOpDeterminismExceptionsTest(test.TestCase):
+  """Test d9m-unimplemented exceptions from CropAndResizeBackprop{Image|Boxes}.
+
+  Test that tf.errors.UnimplementedError is thrown or not thrown, as
+  appropriate, by the GPU code-paths for CropAndResizeBackprop{Image|Boxes} when
+  deterministic ops are enabled.
+
+  This test assumes that the base op test runs all the same test cases when
+  deterministic ops are not enabled and will therefore detect erroneous
+  exception throwing in those cases.
```

Comment:
I don't quite understand this comment. What is the "base op test"?

Reply:
The upcoming commit, changes "base op test" to "test_base.CropAndResizeOpTestBase" in this particular instance. It also makes various similar clarifications elsewhere in `image_grad_deterministic_test.py`, `xent_op_deterministic_test.py`, and `sparse_xent_op_deterministic_test.py`.